crowbar_framework/lib/crowbar/checks.rb

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "resolv"
+require "ipaddr"
+require "socket"
+require "logger"
+
+module Crowbar
+  class NetworkChecks
+
+    attr_accessor :logger
+
+    class << self
+      def ip_addrs(fqdn, version = "ipv4")
+        ip_addrs = []
+
+        Resolv.getaddresses(fqdn).each do |address|
+          ip_addr = IPAddr.new(address)
+          if version == "ipv6"
+            ip_addrs.push address if ip_addr.ipv6?
+          else
+            ip_addrs.push address if ip_addr.ipv4?
+          end
+        end
+
+        ip_addrs
+      end
+    end
+
+    def initialize(logger = ::Logger.new(STDOUT))
+      self.logger = logger
+    end
+
+    def run_all_checks
+      [
+        :fqdn_detected?,
+        :ip_address_resolved?,
+        :loopback_resolved?,
+        :ip_configured?,
+        :network_json_valid?,
+        :firewall_disabled?,
+        :ping_succeeds?
+      ].each do |check|
+        return false unless send(check)
+      end
+
+      true
+    end
+
+    def fqdn_detected?
+      if hostname.nil? || domain.nil?
```

Comment:
You're actually not checking the fqdn here, so really add `|| fqdn.nil?`.

Also since we have a proper language, we might be able to improve the validation of fqdn by checking that it's more than just a hostname.